cli/azd/pkg/ux/confirm.go

```diff
@@ -0,0 +1,259 @@
+package ux
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/ux/internal"
+
+	"dario.cat/mergo"
+	"github.com/eiannone/keyboard"
+	"github.com/fatih/color"
+)
+
+// ConfirmOptions represents the options for the Confirm component.
+type ConfirmOptions struct {
+	// The writer to use for output (default: os.Stdout)
+	Writer io.Writer
+	// The reader to use for input (default: os.Stdin)
+	Reader io.Reader
+	// The default value to use for the prompt (default: nil)
+	DefaultValue *bool
+	// The message to display before the prompt
+	Message string
+	// The optional message to display when the user types ? (default: "")
+	HelpMessage string
+	// The optional hint text that display after the message (default: "[Type ? for hint]")
+	Hint string
+	// The optional placeholder text to display when the value is empty (default: "")
+	PlaceHolder string
+}
+
+var DefaultConfirmOptions ConfirmOptions = ConfirmOptions{
+	Writer: os.Stdout,
+	Reader: os.Stdin,
+}
+
+// Confirm is a component for prompting the user to confirm a message.
+type Confirm struct {
+	canvas Canvas
+	input  *internal.Input
+
+	options            *ConfirmOptions
+	hasValidationError bool
+	value              *bool
+	showHelp           bool
+	complete           bool
+	submitted          bool
+	displayValue       string
+	cancelled          bool
+	cursorPosition     *CursorPosition
+}
+
+// NewConfirm creates a new Confirm instance.
+func NewConfirm(options *ConfirmOptions) *Confirm {
+	mergedOptions := ConfirmOptions{}
+	if err := mergo.Merge(&mergedOptions, options, mergo.WithoutDereference); err != nil {
+		panic(err)
+	}
+
+	if err := mergo.Merge(&mergedOptions, DefaultConfirmOptions, mergo.WithoutDereference); err != nil {
+		panic(err)
+	}
```

Comment:
I don't think we need to bring the mergo library for what we need here. 
Since the options arg is a pointer, we just need to either use it (when different than nil) or not (when nil we create and use the default options).

Reply:
The goal here was to support the user on only setting the options that they really need and allowing the remaining options to be set using the component defaults vs them having to manually constructing a fully valid options object.

I've taken the approach of making any optional options accept `nil` values to help support this concept.